### PR TITLE
Added verification for account states for 'total_funds' property test

### PR DESCRIPTION
**Pull request by dkijania**

### CODE REVIEW

File: chain-addr/src/lib.rs [hunk 1]
```diff
@@ -54,7 +54,7 @@ cfg_if! {
 // Allow to differentiate between address in
 // production and testing setting, so that
 // one type of address is not used in another setting.
-#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[derive(Debug, Clone, Copy, PartialEq, PartialOrd, Eq, Hash)]
```
Comment (mzabaluev) [inline chain-addr/src/lib.rs]: Why are these `PartialOrd` impls needed? It's unusual to add ordering to enums that don't have a business logic order.

File: chain-impl-mockchain/src/testing/address.rs [hunk 2]
```diff
@@ -108,7 +135,7 @@ impl AddressData {
     }
 }
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, PartialEq, PartialOrd)]
```
Comment (mzabaluev) [inline chain-impl-mockchain/src/testing/address.rs]: Good for `PartialEq`, dodgy for `PartialOrd`.

File: chain-impl-mockchain/src/testing/arbitrary/transaction.rs [hunk 3]
```diff
@@ -166,6 +166,10 @@ impl ArbitraryValidTransactionData {
         self.addresses.iter().map(|x| x.make_output()).collect()
     }
 
+    pub fn addresses(&self) -> Vec<AddressDataValue> {
+        self.addresses.iter().cloned().collect()
```
Comment (mzabaluev) [inline chain-impl-mockchain/src/testing/arbitrary/transaction.rs]: What's the point if `self.addresses` is public?

File: chain-impl-mockchain/src/testing/arbitrary/transaction.rs [hunk 4]
```diff
@@ -181,3 +185,75 @@ impl ArbitraryValidTransactionData {
...
+}
+
+
+impl AccountStatesVerifier {
+
+    pub fn new(transaction_data: ArbitraryValidTransactionData) -> Self {
+        AccountStatesVerifier(transaction_data)
+    }
+
+    fn find_equal_and_sub(x: AddressDataValue, collection:  &Vec<AddressDataValue>) -> AddressDataValue {
```
Comment (mzabaluev) [inline chain-impl-mockchain/src/testing/arbitrary/transaction.rs]: No need for this to be a method.

File: chain-impl-mockchain/src/testing/arbitrary/transaction.rs [hunk 5]
```diff
@@ -181,3 +185,75 @@ impl ArbitraryValidTransactionData {
...
+        match collection.iter().cloned().find(|y| y.address_data== x.address_data){
+                   Some(y) =>  AddressDataValue::new(x.address_data, (x.value - y.value).unwrap().clone()),
+                   None => x,
+        }
+    }
+
+    fn find_equal_and_add(x: AddressDataValue, collection:  &Vec<AddressDataValue>) -> AddressDataValue {
```
Comment (mzabaluev) [inline chain-impl-mockchain/src/testing/arbitrary/transaction.rs]: Same as above.

File: chain-impl-mockchain/src/testing/arbitrary/transaction.rs [hunk 6]
```diff
@@ -181,3 +185,75 @@ impl ArbitraryValidTransactionData {
...
+        match collection.iter().cloned().find(|y| y.address_data== x.address_data){
+                   Some(y) =>  AddressDataValue::new(x.address_data, (x.value + y.value).unwrap().clone()),
+                   None => x,
+        }
+    }
+
+    fn filter_accounts(x: &AddressDataValue) -> bool {
```
Comment (mzabaluev) [inline chain-impl-mockchain/src/testing/arbitrary/transaction.rs]: Same as above. These all could be free functions in a module.

File: chain-impl-mockchain/src/testing/address.rs (hunk 2 above)
Comment (dkijania) [reply chain-impl-mockchain/src/testing/address.rs]: it's some leftovers from refactoring. removed

**Pull request merged**